logstash-core/lib/logstash/instrument/wrapped_write_client.rb

```diff
@@ -10,7 +10,12 @@ def initialize(write_client, pipeline, metric, plugin)
       @events_metrics = metric.namespace([:stats, :events])
       @pipeline_metrics = metric.namespace([:stats, :pipelines, pipeline_id, :events])
       @plugin_events_metrics = metric.namespace([:stats, :pipelines, pipeline_id, :plugins, plugin_type, plugin.id.to_sym, :events])
-
+      @events_metrics_counter = @events_metrics.counter(:in)
```

Comment:
@jakelandis from the code it seems to me we're safe here. Also, we'd see a bug with these in `master`:

```rb
@plugin_events_metrics = metric.namespace([:stats, :pipelines, pipeline_id, :plugins, plugin_type, plugin.id.to_sym, :events])
```

These metrics are clearly related to the `pipeline_id`, so if those are still fine on reloads and such, then we know `@events_metrics.counter(:in)` in the same block is also called again on reloads and hence any `prune` that happened will show up here.
Not that we only cache the terminal nodes of each `metric` so to say, each call to like `@events_metrics.counter(:in)` will still go to the metrics store and get us a "fresh" counter :)
For new plugins added we're good for that very same reason, since adding new plugins entails a call to `metric.namespace`.